flexmeasures/data/models/planning/tests/test_solver.py

```diff
@@ -1635,3 +1635,107 @@ def test_battery_stock_delta_quantity(add_battery_assets, gain, usage, expected_
         assert all(scheduler_info[5][0]["stock delta"] == expected_delta)
     else:
         assert all(scheduler_info[5][0]["stock delta"].isna())
+
+
+@pytest.mark.parametrize(
+    "efficiency,expected_efficiency",
+    [
+        ("100%", 1),
+        (
+            "110%",
+            1,
+        ),  # value exceeding the upper bound (100%). It clips the values above 100%.
+        (
+            "90%",
+            0.9,
+        ),  # percentage unit to dimensionless units within the [0,1] interval
+        (0.9, 0.9),  # numeric values are interpreted as dimensionless
+        (
+            None,
+            None,
+        ),  # if the `storage-efficiency` is not defined, the constraint dataframe
+        # uses NaN.
```

Comment:
Good point, this is not happening because we are not updating the field in case of not providing an efficiency:

```python
        if isinstance(storage_efficiency, ur.Quantity) or isinstance(
            storage_efficiency, Sensor
        ):
            device_constraints[0]["efficiency"] = (
                get_continuous_series_sensor_or_quantity(
                    quantity_or_sensor=storage_efficiency,
                    actuator=sensor,
                    unit="dimensionless",
                    query_window=(start, end),
                    resolution=resolution,
                    beliefs_before=belief_time,
                    fallback_attribute="storage_efficiency",  # this should become storage-efficiency
                    max_value=1,
                )
                .fillna(1.0)
                .clip(lower=0.0, upper=1.0)
            )
        elif storage_efficiency is not None:
            device_constraints[0]["efficiency"] = storage_efficiency
```